lib/decorators/formatted_dates.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+require 'scraped'
+
+# Turns a date like 1.1.64 into a preferred format: 1964-01-01
+class FormattedDates < Scraped::Response::Decorator
+  def body
+    Nokogiri::HTML(super).tap do |doc|
+      doc.traverse do |node|
+        next unless node.text?
+        node.text.scan(/(\d\d?\.\d\d?\.\d\d(?:\d\d)?)/).each do |m|
+          node.content = node.text.gsub(m.first, formatted_date(m.first))
```

Comment:
Looping over matches like this manually is a little verbose, and I can't see why it's needed. Can you do this with block form of `gsub` instead? — i.e.

```ruby
      doc.traverse do |node|
        next unless node.text?
        node.content = node.text.gsub(/(\d\d?\.\d\d?\.\d{2,4})/) { |match| Date.strptime(match, '%d.%m.%y').to_s }
      end
```

However, I'm also not sure why you're even looping through nodes etc like this at all. Can this whole class be simplified to just:

```ruby
class FormattedDates < Scraped::Response::Decorator
  def body
    super.gsub(/(\d{1,2}\.\d{1,2}\.\d{2,4})/) { |match| Date.strptime(match, '%d.%m.%y').to_s }
  end
end
```